comp/api/api/apiimpl/internal/agent/agent.go

```diff
@@ -99,10 +99,12 @@ func SetupHandlers(
 	r.HandleFunc("/version", common.GetVersion).Methods("GET")
 	r.HandleFunc("/hostname", getHostname).Methods("GET")
 	r.HandleFunc("/stop", stopAgent).Methods("POST")
-	r.HandleFunc("/status", func(w http.ResponseWriter, r *http.Request) { getStatus(w, r, statusComponent) }).Methods("GET")
+	r.HandleFunc("/status", func(w http.ResponseWriter, r *http.Request) {
+		getStatus(w, r, statusComponent, optional.NewNoneOption[string]())
```

Comment:
I think `optional` here might be a bit overkill. Could we pass just a string? 

Id the string is empty it means we fetch the entire status 😄